src/graphql/client/graphql-client.ts

```diff
@@ -455,17 +455,51 @@ export class GraphQLClient {
         ids.push(record.id);
         // delete in batches
         if (ids.length >= this.resetPageSize) {
-          await this.deleteById(model, ids, deleteSessionId);
+          await this.deleteByIdWithConditions(
+            model,
+            ids,
+            deleteConditions,
+            deleteSessionId
+          );
           ids = [];
         }
       }
       // clean up any remaining records
       if (ids.length > 0) {
-        await this.deleteById(model, ids, deleteSessionId);
+        await this.deleteByIdWithConditions(
+          model,
+          ids,
+          deleteConditions,
+          deleteSessionId
+        );
       }
     }
   }
 
+  async deleteByIdWithConditions(
+    model: string,
+    ids: string[],
+    deleteConditions: any,
+    session: string
+  ): Promise<void> {
+    const ctxExpr = this.supportsSetCtx
+      ? `setCtx(args: {session: "${session}"}) { success }`
+      : undefined;
+    const mutation = {
+      ctx: ctxExpr,
```

Comment:
@ted, is there a good way to test when session is supported?

Reply:
Yeah not sure this is gunna work with the jsonToGraphQLQuery tool we use..

Reply:
you'd have to build it up from objects the same way a query is built.

Reply:
> @ted, is there a good way to test when session is supported? I'm not certain this does what I want.

that is what the `this.supportSetCtx` does.  This `true` if `setCtx` is a valid GraphQL construct.